src/crate/theme/vendor/rtd_compat/extension.py

```diff
@@ -1,255 +0,0 @@
-    github_user, github_repo = get_github_username_repo(git_clone_url)
-    bitbucket_user, bitbucket_repo = get_bitbucket_username_repo(git_clone_url)
-    gitlab_user, gitlab_repo = get_gitlab_username_repo(git_clone_url)
-
-    project_slug = os.environ.get("READTHEDOCS_PROJECT")
-    version_slug = os.environ.get("READTHEDOCS_VERSION")
-    production_domain = os.environ.get("READTHEDOCS_PRODUCTION_DOMAIN", "readthedocs.org")
-
-    scheme = "https"
-    if production_domain.startswith("devthedocs"):
-        scheme = "http"
-
-    # We are using APIv2 to pull active versions, downloads and subprojects
-    # because APIv3 requires a token.
-    try:
-        response_project = requests.get(
-            f"{scheme}://{production_domain}/api/v3/projects/{project_slug}/",
-            timeout=2,
-        ).json()
-        language = response_project["language"]["code"]
-    except Exception:
-        logger.warning(
-            "An error ocurred when hitting API to fetch project language. Defaulting to 'en'.",
-            exc_info=True,
-        )
-        language = "en"
-
-    try:
-        response_versions = requests.get(
-            f"{scheme}://{production_domain}/api/v3/projects/{project_slug}/versions/?active=true",
-            timeout=2,
-        ).json()
-        versions = [
-            (version["slug"], f"/{language}/{version['slug']}/")
-            for version in response_versions["results"]
-        ]
-    except Exception:
-        logger.warning(
-            "An error ocurred when hitting API to fetch active versions. Defaulting to an empty list.",
-            exc_info=True,
-        )
-        versions = []
-
-    try:
-        downloads = []
-        for version in response_versions["results"]:
-            if version["slug"] != version_slug:
-                continue
-
-            for key, value in version["downloads"]:
-                downloads.append(
-                    (
-                        key,
-                        value,
-                    ),
-                )
-    except Exception:
-        logger.warning(
-            "An error ocurred when generating the list of downloads. Defaulting to an empty list.",
-            exc_info=True,
-        )
-        downloads = []
-
-    try:
-        subprojects = []
-        response_project = requests.get(
-            f"{scheme}://{production_domain}/api/v2/project/?slug={project_slug}",
-            timeout=2,
-        ).json()
-        project_id = response_project["results"][0]["id"]
-
-        response_subprojects = requests.get(
-            f"{scheme}://readthedocs.org/api/v2/project/{project_id}/subprojects/",
-            timeout=2,
-        ).json()
-        for subproject in response_subprojects["subprojects"]:
-            subprojects.append(
-                (
-                    subproject["slug"],
-                    subproject["canonical_url"],
-                ),
-            )
```

Comment:
We are currently not using subprojects, so we can think about removing this snippet, in the same spirit to fire less requests to the RTD API?